File: backend/TuneVault.Domain/Entities/Playlist.cs
using System;
using System.Collections.Generic;

namespace TuneVault.Domain.Entities;

public class Playlist
{
    public Guid Id { get; set; }
    public string Title { get; set; } = "";
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
    public bool IsPublic { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid UserId { get; set; }
    public ApplicationUser User { get; set; } = null!;

    public ICollection<PlaylistTrack> PlaylistTracks { get; set; } = [];
}

File: frontend/src/types/playlist.ts
import type { MediaItem } from './media';

export interface Playlist {
  id: string;
  title: string;
  description?: string;
  coverImageUrl: string | null;
  isPublic?: boolean;
  userId?: string;
  createdAt?: string;
  
  // Mảng chứa các bài hát thuộc playlist này (sẽ có dữ liệu khi gọi API GetPlaylistDetail)
  tracks?: MediaItem[]; 
}